src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/IFileOriginsPageViewModel.cs

```diff
@@ -8,7 +7,9 @@ namespace NexusMods.App.UI.Pages.ModLibrary;
 
 public interface IFileOriginsPageViewModel : IPageViewModelInterface
 {
-    LoadoutId LoadoutId { get; set; }
+    LoadoutId LoadoutId { get; }
     
     ReadOnlyObservableCollection<IFileOriginEntryViewModel> FileOrigins { get; }
+    
+    void Initialize(LoadoutId loadoutId);
```

Comment:
Don't do this. Just make `LoadoutId` reactive and move the implementation into a `WhenAnyValue` statement.

Reply:
I don't see the advantage. `LoadoutId` is only ever going to be set once, and that should only happen at initialization.
This code better represents what is going to happen and what is expected. 

WhenAnyValue would add the overhead of a subscription, and would be less expressive, so I thought this approach would be better.

Reply:
Then put `LoadoutId` as a parameter into the constructor. That's what the constructor is for, it initializes the class. Having a method called `Initialize` that does class initialization is literally the job of a constructor. Obviously wouldn't be able to use DI, and have to manually constructor the class, but it's still better than this evil.

Reply:
Not using DI isn't nice either. This pattern isn't uncommon for DI scenarios and for Async initialization. 
It's a tradeoff, I don't know if this is so bad to be worth changing.

Reply:
It's an antipattern.

- Constructor: no overhead, can't create VM using DI, usage is similar to other VMs
- Reactive Property: tiny overhead, create VM using DI, usage is the same as other VMs
- Initialize Method: tiny overhead, create VM using DI, usage is different to other VMs

With a reactive property, you can take this VM everywhere, and it follows the same patterns as everything else we have in our code (we do use `Initialize` in some places and I really want to make those disappear as well).

Reply:
Done